### PR TITLE
W-11558572 - Added tests for cross-spec validation and updated OAS Co…

### PR DESCRIPTION
**Pull request by looseale**
…mponent test

### CODE REVIEW

File: amf-cli/shared/src/test/resources/cross-spec/oas3-point-oas2/api-pointed.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          "type": "string"
+        },
+        "a2": {
+          "type": "integer"
+        }
+      }
+    }
+    }
+  }
+}
```
Comment (pope1838) [inline amf-cli/shared/src/test/resources/cross-spec/oas3-point-oas2/api-pointed.json:21]: Final "}" is an Invalid token

File: amf-cli/shared/src/test/scala/amf/components/OasComponentLinkingTest.scala [hunk 2]
```diff
@@ -60,15 +62,12 @@ class OasComponentLinkingTest extends AsyncFunSuite with Matchers {
 
   private def withComponent(uri: String): Future[Module] = {
     val errorHandler: ErrorHandlerProvider = () => UnhandledErrorHandler
-    OASConfiguration
-      .OAS30()
```
Comment (pope1838) [inline amf-cli/shared/src/test/scala/amf/components/OasComponentLinkingTest.scala:64]: Why not OasConfiguration.Oas30Component()?

File: amf-cli/shared/src/test/resources/cross-spec/oas3-point-oas2/api-pointed.json (hunk 1 above)
Comment (looseale) [reply amf-cli/shared/src/test/resources/cross-spec/oas3-point-oas2/api-pointed.json:21]: Fixed

File: amf-cli/shared/src/test/scala/amf/components/OasComponentLinkingTest.scala (hunk 2 above)
Comment (looseale) [reply amf-cli/shared/src/test/scala/amf/components/OasComponentLinkingTest.scala:64]: Moved to OasConfiguration

**Review by tomsfernandez — DISMISSED** [review]

**Review by nschejtman — APPROVED** [review]

**Pull request merged**